inst/www/shared/shiny.js

```diff
@@ -1413,13 +1413,26 @@ $.extend(imageOutputBinding, {
     var outputId = this.getId(el);
 
     var $el = $(el);
-    // Load the image before emptying, to minimize flicker
-    var img = null;
+    var canvas;
 
     // Remove event handlers that were added in previous renderValue()
     $el.off('.image_output');
-    // Trigger custom 'remove' event for any existing images in the div
-    $el.find('img').trigger('remove');
+
+    // Get existing canvas element if present.
+    var $canvas = $el.find('canvas');
+
+    if ($canvas.length === 0) {
+      // If a canvas element is not already present, that means this is either
+      // the first time renderValue() has been called, or this is after an
+      // error.
+      canvas = document.createElement('canvas');
+      $el.append(canvas);
+      $canvas = $(canvas);
+    } else {
+      // Trigger custom 'reset' event for any existing images in the div
+      canvas = $canvas[0];
+      $canvas.trigger('reset');
+    }
 
     if (!data) {
       $el.empty();
```

Comment:
It looks like I added this block in aed31c0eba9481b05ae5412a4271a452970fc541, but I actually don't know/remember the circumstances in which it will reach this line of code. I may have thought at the time that it would get here if there was an error, but that's not correct -- it doesn't get there if there's an error, since `renderError` gets called instead of `renderValue`.

Am I (now) missing some situation where it would get here?